FlooringOrderingSystem/FlooringOrderingSystem.Data/ProductionOrders.cs

```diff
@@ -0,0 +1,350 @@
+        {
+            DisplayAllOrdersResponse response = new DisplayAllOrdersResponse();
+
+            Order order = new Order();
+
+            
+            bool OrderFileExists = CheckOrderFileExist(OrderDate, out string filepath);
+            if (!OrderFileExists)
+            {
+
+                response.Success = false;
+                response.Message = "invalid order date";
+                response.Orders = null;
+                return response;
+
+            }
+
+            using (StreamReader reader = new StreamReader(filepath))
+            {
+                string line = reader.ReadLine();
+                while ((line = reader.ReadLine()) != null)
+                {
+                    string[] columns = line.Split(',');
+                    order.OrderNumber = int.Parse(columns[0]);
+                    order.OrderDate = OrderDate;
+                    order.CustomerName = AddCommaintheData(columns[1]);
+                    order.State = columns[2];
+                    order.TaxRate = decimal.Parse(columns[3]);
+                    order.ProductType = columns[4];
+                    order.Area = decimal.Parse(columns[5]);
+                    order.CostPerSquareFoot = decimal.Parse(columns[6]);
+                    order.LaborCostPerSquareFoot = decimal.Parse(columns[7]);
+                    order.MaterialCost = decimal.Parse(columns[8]);
+                    order.LaborCost = decimal.Parse(columns[9]);
+                    order.Tax = decimal.Parse(columns[10]);
+                    order.Total = decimal.Parse(columns[11]);
+
+
+
+                    orderList.Add(order);
+                    order = new Order();
+
+
+                }
+
+            }
+
+            response.Orders = orderList;
+            response.Success = true;
+            response.Message = "Order Data for the date is available";
+
+            return response;
+        }
+
+        public SaveOrderResponse SaveOrder(Order Order)
+        {
+            SaveOrderResponse response = new SaveOrderResponse();
+
+            List<string> FileLines = new List<string>();
+            string newline;
+            bool _orderFound = false;
+            int _highestOrderNumber = 0;
+
+
+            bool OrderFileExists = CheckOrderFileExist(Order.OrderDate, out string filepath);
+            if (!OrderFileExists)
+            {
+                newline = BuildFileHeaderLine();
+                FileLines.Add(newline);
+
+                Order.OrderNumber = 1;  //assign 1 to the order number since it's the first one for the order date
+                newline = BuildFileLineFromOrder(Order);
+                FileLines.Add(newline);
+
+                File.WriteAllLines(filepath, FileLines);
+                response.Success = true;
+                response.Message = "Order saved";
+                response.Order = Order;
+
+                return response;
+            }
+
+
+            using (StreamReader reader = new StreamReader(filepath))
```

Comment:
Why do you need to read the file after saving it?

Reply:
SaveOrderResponse handles saving data for all situations.
1. Saving Order when its the first order for the day - so create file, save the first order and close
2. Adding a new order in an existing order file - read the file, find the highest order number, add the order with order number = 1+ highest order,close the file
3. Updating the order with the new info = read the file,find the existing order, replace the existing order with the updated order and close the file.
Hence, you are seeing the code to read the order file.